shared/util/saga.js

```diff
@@ -128,8 +128,8 @@ function* sequentially(effects: Array<any>): Generator<any, Array<any>, any> {
 function safeTakeEveryPure<A, R, FinalAction, FinalActionError>(
   pattern: string | Array<any> | Function,
   pureWorker: ((action: A, state: TypedState) => any) | ((action: A) => any),
-  actionCreatorsWithResult?: (result: R) => FinalAction,
-  actionCreatorsWithError?: (result: R) => FinalActionError
+  actionCreatorsWithResult?: (result: R, action: A) => FinalAction,
```

Comment:
I think its worth exposing this to these helpers. I know you can emit a Saga.all([]) etc to encode this info but i think there's a lot of cases where you want this